### DISCUSSION TITLE
Injecting ILoggerFactory for test output

### DISCUSSION THREAD

**Discussion by apazureck** [AutoFixture/AutoFixture#1240] [Q&A]
Hi,

First of all: Thank you for that great project. Saves me a ton of time.

I am doing a lot of logging in my classes using the logging abstractions from Microsoft using `XUnit` and the `Divergic.Logging.Xunit` Nuget packages. So I wrote a customization to automatically inject my `ILoggerFactory` to have my logs on the test output.

I would like to publish that package on nuget providing an extension method for `IFixture` which injects the loggers using the provided `ILoggerFactory`.

You can have a look at the repo here:

https://github.com/apazureck/AutoFixture.Customizations.LoggerFactoryCustomization/blob/master/README.md

Would you be OK doing it that way - and if not: How should I publish this extension to AutoFixture?

Cheers

**Comment by aivascu** [comment]
Hello @apazureck ,
Thank you, for contacting us. We welcome contributions from the community.

I've had a look at your repository and there are a couple things I'd like to mention.

First of all, we have reserved on [nuget.org](https://nuget.org) all `AutoFixture.*` package namespaces, except for `AutoFixture.Community.*` which we reserved for community packages. Considering this you might want to change the name of the project.

Next is the naming convention. I would recommend using naming conventions consistent to those used in `AutoFixture` packages.
There a couple places, in your codebase, where the namespaces are ambiguous with names of abstractions. For example `LoggerFactoryCustomization` is both a namespace and a class. Also the class library project name ends with something that looks like the name of an interface (i.e. `Autofixture.Customizations.ILoggerFactory` ), this might be confusing to contributors.
IMO it would be better if you used a name like `AutoFixture.Community.Logging.Abstractions` or  `AutoFixture.Community.Microsoft.Logging` to avoid conflicts or ambiguities.

Finally you might want to depend on the earliest major version of `AutoFixture`, that you can use for your library, to ensure compatibility with as many package versions as possible.

**Reply by apazureck** [reply]
Great, thank you for the feedback, that's why I asked ;)

I'll make issues for the things you pointed out and come back when I'm done, so you can double check.

Cheers!
